[TH2-846] Rewrite generator for gRPC interfaces. Add python services generation (#1)
th2-net/th2-grpc-service-generator: README.md

# Overview

> This project implemented protoc plugin for generation services for gRPC router.

Reviewer: ```suggestion
This project implements a **protoc** plugin for generation services for gRPC router.
```

> It generates java and python code.

Reviewer: ```suggestion
It generates wrappers for Java's and Python's gRPC implementations.
```

You can plug it to protoc or run from command line.
# Usage
## Protoc plugin

> If you want to use this generator with gradle and protoc you should add artifact name to protoc plugins section and add to generated protoc tasks

Reviewer: ```suggestion
If you want to use this generator with Gradle and the protoc plugin you should add the project's artifact to the protoc plugin's section and add the required options to the proto generation tasks
```

> You can set options:

Reviewer: ```suggestion
You can set the following options:
```

> 1. javaInterfacesPath - generate services interfaces to this path
> 1. javaInterfacesImplPath - generate TH2 services interfaces implementation to this path
> 1. pythonPath - generate python services to this path

Reviewer: ```suggestion
1. javaInterfacesPath - the path for Java services interfaces
1. javaInterfacesImplPath - the path for TH2 services implementations
1. pythonPath - the path for Python services implementations
```

> All these paths will be relative from  ``generateProtoTasks.generatedFilesBaseDir`` (Example: `javaInterfacesPath` = `path/to/base/dir/path/to/interfaces`)

Reviewer: Also, please add information about the default behaviour (when neither of those options is set)

```suggestion
All these paths are relative to the directory in  ``generateProtoTasks.generatedFilesBaseDir`` parameter (Example: `javaInterfacesPath` = `path/to/base/dir/path/to/interfaces`)
```

```groovy
protobuf {
    plugins {
        generator {
            artifact = "com.exactpro.th2:service-generator-protoc:${version_generator}:all@jar"
        }
    }
    generateProtoTasks.generatedFilesBaseDir = 'path/to/base/dir'
    generateProtoTasks {
        all()*.plugins {
            generator {
                option 'javaInterfacesPath=./path/to/interfaces'
                option 'javaInterfacesImplPath=./path/to/th2/impl'
                option 'pythonPath=./path/to/python'
            }
        }
    }

> }

Reviewer: What about calling default grpc generation?
```groovy
    grpc {
        artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}"
    }
       // ...
    generateProtoTasks {
        all()*.plugins {
              grpc {}
        }
    }
```

```
## Command line
You can run this project with command line

> First argument should be directory or file with FileDescriptorSet. Second argument is output directory. After that arguments you can set options (Example: `--pythonPath=./path/to/python`)

Reviewer: Please, clarify what FileDescriptorSet means. Also, it would be great to describe how and where user can get the artifact to run the generation from cmd
